docs/architecture.rst

```diff
@@ -149,66 +155,85 @@ if any of the environment variables *SNF_IMAGE_DEV*,
 will fail if the Linux distribution is not supported and ``/etc/hostname`` is
 not present on the file system. For now, we support Debian, Red Hat, Slackware,
 SUSE and Gentoo derived distributions. The hostname is read from
 *SNF_IMAGE_HOSTNAME* variable. In addition to the latter, *SNF_IMAGE_TARGET* is
-also required.
+also required. For cloud-init, the configuration is performed by setting the
+*local-hostname* key of the datasource's metadata.
 
 **ChangeMachineId**: On Linux instances, this script will generate a new random
 machine ID and will place it in ``/etc/machine-id``. For more info check
 `here <https://www.freedesktop.org/software/systemd/man/machine-id.html>`_. The
 task will fail if *SNF_IMAGE_TARGET* is missing.
 
-**ChangePassword**: Changes the password for a list of existing users. On Linux 
-systems this is accomplished by directly altering the instance's
-``/etc/shadow`` file. On Windows systems a script is injected into the VM's
-hard disk. This script will be executed during the specialize pass of the
-Windows setup. On \*BSD systems ``/etc/master.passwd`` is altered,
+**ChangePassword**: Changes the authentication credentials for a list of
+existing users. On Linux systems this is accomplished by directly altering the
+instance's ``/etc/shadow`` file. On Windows systems a script is injected into
+the VM's hard disk. This script will be executed during the specialize pass of
+the Windows setup. On \*BSD systems ``/etc/master.passwd`` is altered,
 ``/etc/spwd.db`` is removed and a script is injected into the VM's hard disk
 that will recreate the aforementioned file during the first boot. The list of
 users whose passwords will changed is determined by the
-*SNF_IMAGE_PROPERTY_USERS* variable (see :ref:`image-properties`). For this
-task to run *SNF_IMAGE_TARGET* and *SNF_IMAGE_PASSWD* variables need to be
-present.
+*SNF_IMAGE_PROPERTY_USERS* variable (see :ref:`image-properties`). On Unix
+systems, if the variable *SNF_IMAGE_AUTH_KEYS* is set, the content of this
+variable is injected to the authorized keys file of each user. For this task to
+run *SNF_IMAGE_TARGET* and *SNF_IMAGE_PASSWD* variables need to be present. For
+cloud-init, the configuration is performed using the *ssh_pwauth* and
+*chpasswd* keys of the *Set Password* module, as well as, the
+*public-keys* key of the datasource's meta-data.
 
 **ConfigureNetwork**: Edit the OS's native network configuration files to
 configure the instance's NICs. This works for most Linux and all the supported
 \*BSD systems. In order to do this, all the NIC_* Ganeti provided environment
 variables are exported to the task. The only variable required by this task is
 *SNF_IMAGE_TARGET*. For this task to work correctly, the user may need to
 adjust the *DHCP_TAGS* and the *\*_DHCPV6_TAGS* configuration parameters (see
-:doc:`/configuration`).
+:doc:`/configuration`). When working with cloud-init enabled images, this task
```

Comment:
Well cloud-init is a service. There are cloud images and cloud-init enabled images but no cloud-init images.